### PR TITLE
Update Dashboard and Alert headers with the Query one

### PR DESCRIPTION
**Pull request by gabrieldutra**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
Following #4694, this PR updates Dashboard and Alert headers to behave similar to the Query Page one (e.g: wrapping the controls)

## Related Tickets & Documents
#2796 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

**Dashboard Page**
| Before | After |
|---------|-----------|
| dashboard-header-before | dashboard-header-after |

**Alert Page**
| Before | After |
|---------|-----------|
| alert-header-before | alert-header-after |

### CODE REVIEW

File: client/app/assets/less/inc/alert.less [hunk 1]
```diff
@@ -1,53 +1,45 @@
...
+  }
 }
-
-.alert-actions {
-    flex-grow: 1;
-    display: flex;
-    justify-content: flex-end;
-    align-items: center;
-    margin-right: -15px;
-}
```
Comment (gabrieldutra) [inline client/app/assets/less/inc/alert.less]: From this file I just moved this to `Title.less`, the rest is on Prettier.

File: client/app/assets/less/redash/query.less [hunk 2]
```diff
@@ -122,23 +122,6 @@ body.fixed-layout {
...
-    transition: opacity 0.2s ease-in-out;
-  }
-
-  &:hover {
-    .tags-control a {
-      opacity: 1;
-    }
-  }
-}
-
```
Comment (gabrieldutra) [inline client/app/assets/less/redash/query.less]: Not used anymore.

File: client/app/assets/less/redash/query.less [hunk 3]
```diff
@@ -517,12 +500,3 @@ nav .rg-bottom {
...
 }
-
-// Responsive fixes
-@media (max-width: 767px) {
-  .query-page-wrapper {
-    h3 {
-      font-size: 18px;
-    }
-  }
-}
```
Comment (gabrieldutra) [inline client/app/assets/less/redash/query.less]: Moved to `QueryPageHeader.less`

File: client/app/pages/dashboards/components/DashboardHeader.jsx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+              <Button
+                className="icon-button m-l-5"
+                type={buttonType(dashboard.publicAccessEnabled)}
+                onClick={showShareDashboardDialog}
+                data-test="OpenShareForm">
+                <i className="zmdi zmdi-share" />
+              </Button>
+            </Tooltip>
+          )}
+          {showMoreOptionsButton && <DashboardMoreOptionsButton dashboardOptions={dashboardOptions} />}
```
Comment (gabrieldutra) [inline client/app/pages/dashboards/components/DashboardHeader.jsx]: Now that the controls are wrapped for mobiles, I'm not hiding those buttons anymore (I saw no reason to keep doing it)

File: client/app/pages/dashboards/components/DashboardHeader.less [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    & > div {
+      padding: 5px 0;
+    }
+
+    h3 {
+      margin: 0;
+
+      @media (max-width: 767px) {
+        font-size: 18px;
```
Comment (gabrieldutra) [inline client/app/pages/dashboards/components/DashboardHeader.less]: Copied this behavior from the Query Page to have more room for tags.

File: client/app/pages/queries/components/QueryPageHeader.less [hunk 6]
```diff
@@ -2,7 +2,7 @@
   display: flex;
   flex-wrap: wrap;
   align-items: stretch;
-  margin: -5px 0;
+  margin-bottom: -5px;
```
Comment (gabrieldutra) [inline client/app/pages/queries/components/QueryPageHeader.less]: This change keeps a margin on the top of the Query Page (to be more consistent with other pages).

File: client/app/pages/dashboards/components/DashboardHeader.jsx (hunk 4 above)
Comment (arikfr) [reply client/app/pages/dashboards/components/DashboardHeader.jsx]: Let's continue hiding the full screen button as it's meaningless on mobile.

Comment (gabrieldutra) [reply client/app/pages/dashboards/components/DashboardHeader.jsx]: That one is still hidden, only the Share and the Dropdown were changes.

**Pull request merged**